### PR TITLE
[Calcite-1665] Aggregates and HAVING cannot be combined

### PR DESCRIPTION
**Pull request by Zhiqiang-He**

### CODE REVIEW

**Comment by julianhyde** [comment]
Looks good (although I haven't run the tests).

Can you keep the indentation the same? Your indentation change might be introducing spurious diffs.

**Comment by Zhiqiang-He** [comment]
indentation  already change in test case . please review it. thanks.

File: core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java [hunk 1]
```diff
@@ -133,11 +133,27 @@ public Result visit(Join e) {
...
+        builder.clauses.add(Clause.HAVING);
+      } else {
+        builder = x.builder(e, Clause.HAVING);
+      }
+      builder.setHaving(builder.context.toSql(null, e.getCondition()));
+      return builder.result();
+    } else {
+      final Builder builder = x.builder(e, Clause.WHERE);
+      builder.setWhere(builder.context.toSql(null, e.getCondition()));
+      if (input instanceof Join) {
```
Comment (jbalint) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: Why did you add this condition and new `result(boolean)` method? All tests are passing without it.

Comment (Zhiqiang-He) [inline core/src/main/java/org/apache/calcite/rel/rel2sql/RelToSqlConverter.java]: Sorry, It's my mistake, I am alias test on reltosql for calcite now.
this function will be removed.

**Comment by Zhiqiang-He** [comment]
@jbalint  Please review it . thanks.
